ipatests/test_integration/test_acme.py

```diff
@@ -0,0 +1,236 @@
+
+    We currently test:
+
+        * service enable/disable (using Curl)
+        * http-01 challenge with Certbot's standalone HTTP server
+        * dns-01 challenge with Certbot and FreeIPA DNS via hook scripts
+        * revocation with Certbot
+        * http-01 challenge with mod_md
+
+    Tests we should add:
+
+        * dns-01 challenge with mod_md (see
+          https://httpd.apache.org/docs/current/mod/mod_md.html#mdchallengedns01)
+
+    Things that are not implmented/supported yet, but may be in future:
+
+        * IP address SAN
+        * tls-alpn-01 challenge
+        * Other clients or service scenarios
+
+    """
+    num_replicas = 0
+    num_clients = 1
+
+    @classmethod
+    def install(cls, mh):
+        # cache the acme service uri
+        acme_host = f'{IPA_CA_RECORD}.{cls.master.domain.name}'
+        cls.acme_server = f'https://{acme_host}/acme/directory'
+
+        # install packages before client install in case of IPA DNS problems
+        if not skip_certbot_tests:
+            cls.clients[0].run_command(['dnf', 'install', '-y', 'certbot'])
+        if not skip_mod_md_tests:
+            cls.clients[0].run_command(['dnf', 'install', '-y', 'mod_md'])
+
+        tasks.install_master(cls.master, setup_dns=True)
+
+        tasks.install_client(cls.master, cls.clients[0])
+        tasks.config_host_resolvconf_with_master_data(
+            cls.master, cls.clients[0]
+        )
+
+    #######
+    # kinit
+    #######
+
+    def test_kinit_master(self):
+        # Some tests require executing ipa commands, e.g. to
+        # check revocation status or add/remove DNS entries.
+        # Preemptively kinit as admin on the master.
+        tasks.kinit_admin(self.master)
+
+    #####################
+    # Enable ACME service
+    #####################
+
+    def test_acme_service_not_yet_enabled(self):
+        # --fail makes curl exit code 22 when response status >= 400.
+        # ACME service should return 503 because it was not enabled yet.
+        self.clients[0].run_command(
+            ['curl', '--fail', self.acme_server],
+            ok_returncode=22,
+        )
+
+    def test_enable_acme_service(self):
+        self.master.run_command(['ipa-acme-manage', 'enable'])
+
+        # wait a short time for Dogtag ACME service to observe config
+        # change and reconfigure itself to service requests
+        time.sleep(3)
+
+        # should succeed now
+        self.clients[0].run_command(['curl', '--fail', self.acme_server])
+
+    ###############
+    # Certbot tests
+    ###############
+
+    @pytest.mark.skipif(skip_certbot_tests, reason='certbot not available')
+    def test_certbot_register(self):
+        # service is enabled; registration should succeed
+        self.clients[0].run_command(
+            [
+                'certbot',
+                '--server', self.acme_server,
+                'register',
+                '-m', 'nobody@example.com',
```

Comment:
Perhaps use test instead of com for the TLD just in case.

Reply:
Sure.  Fixed in next push.